flight/Modules/Stabilization/stabilization.c

```diff
@@ -941,13 +976,25 @@ static void SettingsUpdatedCb(UAVObjEvent * ev)
 		// The dT has some jitter iteration to iteration that we don't want to
 		// make thie result unpredictable.  Still, it's nicer to specify the constant
 		// based on a time (in ms) rather than a fixed multiplier.  The error between
-		// update rates on OP (~300 Hz) and CC (~475 Hz) is negligible for this
-		// calculation
+		// update rates on OP (~300 Hz) and CC (~475 Hz) is negligible for this calculation
+		//fakeDt now only used for vbar_decay, for gyro filter now a real, but filtered, dT is used
 		const float fakeDt = 0.0025f;
-		if(settings.GyroTau < 0.0001f)
-			gyro_alpha = 0;   // not trusting this to resolve to 0
+
+		//TODO: if the infrastructure for alarms will be more advanced in the future,
+		//      we should assert a alarm here when GyroCutoff > 90% of F_nyquist and LPF gets disabled
+
+		if (settings.GyroCutoff < 1.0f) {	// disable LPF when GyroCutoff < 1.0 Hz
+			gyro_filter_enabled = GYRO_LPF_DISABLED;
+			settings.GyroCutoff = 0;
+		}
+		else if ( (dT > 0) && (settings.GyroCutoff > (0.45f / dT)) ) {	// disable LPF when GyroCutoff > 90% of F_nyquist (= Fs/2 = 1/(2*dT)); using here not the filtered dT, because this is at the start to low
+			gyro_filter_enabled = GYRO_LPF_DISABLED;
+			settings.GyroCutoff = 0;
+			StabilizationSettingsGyroCutoffSet(&settings.GyroCutoff); // Set GyroCutoff to 0 to signalize in GCS, that it is disabled
```

Comment:
Sorry , i was not aware we generally don't like the practice of altering the settings from the FC in taulabs.
I thought i had already seen functionality like that somewhere in the code (but maybe it was OP or something else), and that was where i got the idea of doing it like that. But I am not 100% sure about it.

Is was aware that when you set again a UAVObject from a settings update Callback, you should avoid triggering endless loops. And I think that should not happen with this implementation.

It seems now like a principal decision is needed how that should be done. And I will than implement the preferred solution.

One other solution might be just removing the feedback about disabling the LPF.

Are there some examples how the alarm/warning system should be used?
Can just variables be added to the alarm UAVObject (/shared/uavobjectdefinition/systemalarms.xml) and than set if a problem is detected?
There seems to be the possibility to set access level (read/write) separate for gcs and flight for the complete UAVObject. Is there also the possibility to set the access level for single elements?